js/map.js

```diff
@@ -0,0 +1,106 @@
+(function () {
+  'use strict';
+
+  var titleArray = ['Big cosy flat',
+    'Small uncosy flat',
+    'Huge beautiful palace',
+    'Small horrible palace',
+    'Beautiful guest house',
+    'Ugly inhospitable house',
+    'Cosy bungalow far from the sea',
+    'Uncosy bungalow in water up to your knees'
+  ];
+
+  var propertyTypeArray = ['palace', 'flat', 'house', 'bungalow'];
+  var checkinTimeArray = ['12:00', '13:00', '14:00'];
+  var checkoutTimeArray = ['12:00', '13:00', '14:00'];
+  var featuresArray = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+  var photosArray = ['http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+    'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+    'http://o0.github .io/assets/images/tokyo/hotel3.jpg'
+  ];
+
+  function generateRandomNumbersArray(arraySize, maxNumber, shiftNumber) {
+    if (arraySize < 1 || maxNumber < 1) {
+      return;
+    }
+
+    var randomNumberArray = [];
+
+    while (randomNumberArray.length < arraySize) {
+      var randomNumber = Math.floor(Math.random() * Math.floor(maxNumber)) + shiftNumber || 0;
+      if (randomNumberArray.indexOf(randomNumber) > -1) {
+        continue;
+      }
+      randomNumberArray.push(randomNumber);
+    }
+    return randomNumberArray;
+  }
+
+  function getRandomNumberInRange(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min)) + min;
+  }
+
+  function getTitle(title) {
+    return title[getRandomNumberInRange(0, title.length)];
+  }
+
+  function getSortedRandomStringsArray(arr) {
+    var stringsArray = [];
+    var randomArray = generateRandomNumbersArray(getRandomNumberInRange(1, arr.length), arr.length, 0).sort(function (a, b) {
+      return a - b
+    });
+
+    randomArray.forEach(function (elem) {
+      stringsArray.push(arr[elem]);
+    })
+    return stringsArray;
+  }
+
+  function getRandomStringsArray(arr) {
+    var stringsArray = [];
+    var randomArray = generateRandomNumbersArray(arr.length, arr.length, 0);
+    randomArray.forEach(function (elem) {
+      stringsArray.push(arr[elem]);
+    })
+
+    return stringsArray;
+  }
+
+  function getRandomListingObj(title, type, checkin, checkout, features, photos, avatarNumber) {
+    return {
+      author: {
+        avatar: 'img/avatars/user{{0' + avatarNumber + '}}.png'
```

Comment:
if you take a look at the images in img you see the image format
{{}}- curly braces in task description symbolise a template string you need to replace